librosa/core/notation.py

```diff
@@ -805,9 +810,17 @@ def key_to_notes(key: str, *, unicode: bool = True) -> List[str]:
 
         notes = notes_flat
 
+    if natural:
+        # Apply natural signs to any note which has no other accidentals and does not appear in the scale for key.
+        scale_notes = key_to_degrees(key)
+        for place, note in enumerate(notes):
+            match = NOTE_RE.match(note)
+            if match and match.group('accidental')=='' and __note_to_degree(note) not in scale_notes:
+                notes[place] = match.group('note')+'♮'
```

Comment:
We can skip some unnecessary regex parsing here by early skipping over notes in the scale:
```suggestion
        scale_notes = key_to_degrees(key)
        for place, note in enumerate(notes):
            if __note_to_degree(note) in scale_notes:
                continue
            match = NOTE_RE.match(note)
            if match and match.group('accidental')=='':
                notes[place] = match.group('note')+'♮'
```
I think we might still be doing some redundant work here though, since `__note_to_degree` also does a NOTE_RE.match call.  Maybe there's a clever way to simplify this?

Reply:
I like that you check whether the note is in the scale first.

I tried a solution without regex parsing which relies on checking whether `note` has length 1, and if so, just adding a natural sign to it. This relies on the fact that `note` is guaranteed to match the regex, but perhaps there are some robustness issues with that assumption.

Reply:
> checking whether `note` has length 1, and if so, just adding a natural sign to it.

Oh that's clever! :grin: 

I can't think of any reason why `len(note) == 1` would fail, as long as the notes have already been simplified (eg to rule out a failing throw on `Gb#`).  In unicode, accidentals are not combining symbols, so they will increment the string length.